fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/Wallet.java

```diff
@@ -271,5 +285,17 @@ private String convertStringToPath(String string){
       return null;
     }
   }
+  private void initKeysManager(Context applicationContext)
+      throws IOException, GeneralSecurityException {
+    AesGcmKeyManager.register(true);
+    AeadConfig.register();
+    keysetManager =
+        new AndroidKeysetManager.Builder()
+            .withSharedPref(applicationContext, "POP_KEYSET_2",  "POP_KEYSET_SP_2")
+            .withKeyTemplate(AesGcmKeyManager.rawAes256GcmTemplate())
+            .withMasterKeyUri("android-keystore://POP_MASTER_KEY_2")
+            .build();
+    aead = keysetManager.getKeysetHandle().getPrimitive(Aead.class);
```

Comment:
From what I see you are creating a new Keyset Manager each time you want to import/export the seed in order to encrypt/decrypt it, why so?

Reply:
yes you are right! is better init only one time!